walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,108 @@
 
 package com.google.sps;
 
+import java.util.Collections;
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays; 
+
 
 public final class FindMeetingQuery {
+
+  private final int MINUTES_IN_A_DAY = 1440;
+
+  /**
+   * Returns an {@ArrayList} of {@code TimeRange}s of when an Event can take place.
+   * Runtime: O(1) -> Regardless of the amount of events/attendees, a day will always
+   * have the same number of minutes, which is what the array represents.
+   * @param busyTimes boolean array containing all times where there is a meeting.
+   * @param request Desired duration of the requested event
+   * @return {@code ArrayList} with all possible timeranges for an event to happen.
+   */
+  private ArrayList<TimeRange> findTimeRanges(boolean[] busyTimes, long duration){
+    ArrayList<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+    // Find free time in the array and if they fit the meeting, create the time range
+    int availabilityInterval = 0;
+    for(int minuteOfTheDay = 0; minuteOfTheDay < MINUTES_IN_A_DAY+1; minuteOfTheDay++){
```

Comment:
nit: instead of
```java
minuteOfTheDay < MINUTES_IN_A_DAY+1
```
you could just write 
```java
minuteOfTheDay <= MINUTES_IN_A_DAY
```

Reply:
Since you're actually iterating over busyTimes, maybe just `minuteOfTheDay < len(busyTimes)` (or whatever the length function is in Java).

It's possible that someone in the future changes busyTimes to not be MINUTES_IN_A_DAY long.